packages/augur-api/src/api/Contracts.ts

```diff
@@ -2,28 +2,89 @@ import { GenericAugurInterfaces } from "@augurproject/core";
 import { ContractAddresses } from "@augurproject/artifacts";
 
 export type SomeRepToken<TBigNumber> = GenericAugurInterfaces.ReputationToken<TBigNumber> | GenericAugurInterfaces.TestNetReputationToken<TBigNumber>;
+export type SomeTime<TBigNumber> = GenericAugurInterfaces.Time<TBigNumber> | GenericAugurInterfaces.TimeControlled<TBigNumber>;
 
 export class Contracts<TBigNumber> {
   public augur: GenericAugurInterfaces.Augur<TBigNumber>;
   public universe: GenericAugurInterfaces.Universe<TBigNumber>;
   public cash: GenericAugurInterfaces.Cash<TBigNumber>;
+  public orders: GenericAugurInterfaces.Orders<TBigNumber>;
+  public createOrder: GenericAugurInterfaces.CreateOrder<TBigNumber>;
+  public cancelOrder: GenericAugurInterfaces.CancelOrder<TBigNumber>;
+  public trade: GenericAugurInterfaces.Trade<TBigNumber>;
+  public completeSets: GenericAugurInterfaces.CompleteSets<TBigNumber>;
+  public claimTradingProceeds: GenericAugurInterfaces.ClaimTradingProceeds<TBigNumber>;
+  public time: SomeTime<TBigNumber> | void;
+  public legacyReputationToken: GenericAugurInterfaces.LegacyReputationToken<TBigNumber>;
+
   public reputationToken: SomeRepToken<TBigNumber> | null = null;
   private readonly dependencies: GenericAugurInterfaces.Dependencies<TBigNumber>;
 
   public constructor (addresses: ContractAddresses, dependencies: GenericAugurInterfaces.Dependencies<TBigNumber>) {
     this.dependencies = dependencies;
     this.augur = new GenericAugurInterfaces.Augur<TBigNumber>(dependencies, addresses.Augur);
-    this.universe = new GenericAugurInterfaces.Universe<TBigNumber>(dependencies, addresses.Universe);
+    this.universe = this.universeFromAddress(addresses.Universe);
     this.cash = new GenericAugurInterfaces.Cash<TBigNumber>(dependencies, addresses.Cash);
+    this.orders = new GenericAugurInterfaces.Orders<TBigNumber>(dependencies, addresses.Orders);
+    this.createOrder = new GenericAugurInterfaces.CreateOrder<TBigNumber>(dependencies, addresses.CreateOrder);
+    this.cancelOrder = new GenericAugurInterfaces.CancelOrder<TBigNumber>(dependencies, addresses.CancelOrder);
+    this.trade = new GenericAugurInterfaces.Trade<TBigNumber>(dependencies, addresses.Trade);
+    this.completeSets = new GenericAugurInterfaces.CompleteSets<TBigNumber>(dependencies, addresses.CompleteSets);
+    this.claimTradingProceeds = new GenericAugurInterfaces.ClaimTradingProceeds<TBigNumber>(dependencies, addresses.ClaimTradingProceeds);
+    this.legacyReputationToken = new GenericAugurInterfaces.LegacyReputationToken<TBigNumber>(dependencies, addresses.LegacyReputationToken);
+
+    if (typeof addresses.Time !== "undefined") {
+      this.time = new GenericAugurInterfaces.TimeControlled<TBigNumber>(dependencies, addresses.Time);
+    }
+  }
+
+  public getTime(): SomeTime<TBigNumber> {
```

Comment:
This could just use the method on the Augur contract for getting time. For setting time though this makes sense

Reply:
Yeah I only actually use this for setting the timestamp.